src/alfasim_sdk/result_reader/reader.py

```diff
@@ -295,3 +330,154 @@ def get_profile_metadata(meta: Dict) -> ProfileMetadata:
             get_profile_metadata(profile_metadata)
             for profile_metadata in metadata.profiles.values()
         ]
+
+
+def _non_empty_dict_validator(values_type: type) -> Callable:
+    def validator(inst: Any, attribute: attr.Attribute, value: Any) -> None:
```

Comment:
You've probably thought about this already, but it would be good if these had more specific types. Not sure if `object` instead of `Any` would work, but at least `Callable[[Any, attr.Attribute, Any] None]`?

Reply:
If this is actionable, it applies to the other validators, of course.